### PR TITLE
Decorators WIP

### PR DESCRIPTION
**Pull request by pzuraq**

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -3537,6 +3548,13 @@ <h1>The List and Record Specification Types</h1>
...
     </emu-clause>
 
+    <emu-clause id="sec-map-specification-type">
+      <h1>The Map Specification Types</h1>
+      <p>The <dfn>Map</dfn> type is used to explain the evaluation of key-value based data structures, such as the metadata data structure built up by decorators (see <emu-xref href="#sec-metadata-record-specification-type"></emu-xref>). Values of the Map type are ordered sequences of key/value pairs containing individual values. These sequences may be of any length. The elements of a map may be randomly accessed using key-based indices. For notational convenience a dynamic-key syntax can be used to access Map elements. For example, _publicMetadata_[*"key"*] is shorthand for saying the value associated with *"key"* in the Map _publicMetadata_. If the map does not have a value associated with that key, then it returns ~empty~.</p>
+
+      <p>Maps can be iterated like lists, and are iterated as key/value pairs. If no order is specified, then elements are iterated in the order they were inserted.</p>
+    </emu-clause>
```
Comment (ljharb) [inline spec.html]: why do we need this type at all? couldn't we make a null object (one that's never observed to exist) and use that for the lookups?

File: spec.html [hunk 2]
```diff
@@ -4206,6 +4224,28 @@ <h1>The ClassFieldDefinition Record Specification Type</h1>
               The initializer of the field, if any.
             </td>
           </tr>
+          <tr>
+            <td>
+              [[DecoratorInitializers]]
+            </td>
+            <td>
+              a list of ECMAScript function objects, or ~empty~
```
Comment (ljharb) [inline spec.html:4232]:
```suggestion
              a List of ECMAScript function objects, or ~empty~
```
any reason not to default it to an empty List?

File: spec.html [hunk 3]
```diff
@@ -4206,6 +4224,28 @@ <h1>The ClassFieldDefinition Record Specification Type</h1>
...
+            <td>
+              Additional initializers added by decorators, if any.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[FinisherInitializers]]
+            </td>
+            <td>
+              a list of ECMAScript function objects, or ~empty~
```
Comment (ljharb) [inline spec.html:4243]:
```suggestion
              a List of ECMAScript function objects, or ~empty~
```
same question

File: spec.html [hunk 4]
```diff
@@ -4222,6 +4262,85 @@ <h1>The ClassFieldDefinition Record Specification Type</h1>
...
+      <p>Values of the ClassMethodInitDefinition type are Record values whose fields are defined as by <emu-xref href="#table-classmethodinitdefinition-fields"></emu-xref>. Such values are referred to as <dfn>ClassMethodInitDefinition Record</dfn>s.</p>
+      <emu-table id="table-classmethodinitdefinition-fields" caption="ClassMethodInitDefinition Record Fields">
+        <table>
+          <tbody>
+          <tr>
+            <td>
+              [[Initializers]]
+            </td>
+            <td>
+              a list of ECMAScript function objects, or ~empty~
```
Comment (ljharb) [inline spec.html:4277]:
```suggestion
              a List of ECMAScript function objects, or ~empty~
```
here also

**Comment by pzuraq** [comment]
Moved to https://github.com/tc39/ecma262/pull/2417

**Pull request closed without merging**